omod/src/main/java/org/openmrs/module/aijar/htmlformentry/DSDSProgramSubmissionAction.java

```diff
@@ -0,0 +1,139 @@
+import org.openmrs.module.htmlformentry.FormEntryContext.Mode;
+import org.openmrs.module.htmlformentry.FormEntrySession;
+
+import java.util.*;
+
+import static org.openmrs.api.context.Context.getProgramWorkflowService;
+import static org.openmrs.module.aijar.AijarConstants.GP_DSDM_CONCEPT_ID;
+import static org.openmrs.module.aijar.AijarConstants.GP_DSDM_PROGRAM_UUID_NAME;
+
+/**
+ * Enrolls patients into the TB program
+ */
+public class DSDSProgramSubmissionAction implements CustomFormSubmissionAction {
+    @Override
+    public void applyAction(FormEntrySession session) {
+        Mode mode = session.getContext().getMode();
+        if (!(mode.equals(FormEntryContext.Mode.ENTER) || mode.equals(FormEntryContext.Mode.EDIT))) {
+            return;
+        }
+
+        Patient patient = session.getPatient();
+        Set<Obs> obsList = session.getEncounter().getAllObs();
+        List<PatientProgram> patientPrograms = getActivePatientProgramAfterThisEncounter(patient, null, session.getEncounter().getEncounterDatetime());
+
+        /**
+         * Terminate wen patient is already enrolled in the program selected.
+         */
+        for (PatientProgram patientProgram : patientPrograms) {
+            /**
+             * Check if Same program is enrolled on the same date
+             */
+            if (patientProgram.getProgram() == getProgramByConceptFromObs(obsList) && patientProgram.getDateEnrolled() == session.getEncounter().getEncounterDatetime()) {
+                return;
+            }
+        }
+
+        /**
+         * Completing all DSDM active programs of the patient
+         */
+        if (!patientPrograms.isEmpty()) {
+            for (PatientProgram previousPatientDSDMProgram : patientPrograms) {
+                /**
+                 * Check if program to enroll is greater than
+                 */
+                if (session.getEncounter().getEncounterDatetime().compareTo(previousPatientDSDMProgram.getDateEnrolled()) > 0) {
+                    if (getDSDMPrograms().contains(previousPatientDSDMProgram.getProgram())) {
+                        previousPatientDSDMProgram.setDateCompleted(session.getEncounter().getEncounterDatetime());
+                        getProgramWorkflowService().savePatientProgram(previousPatientDSDMProgram);
+                    }
+                }
+            }
+        }
+
+        /**
+         * Create new PatientProgram
+         */
+        if (getProgramByConceptFromObs(obsList) != null) {
+            PatientProgram newPatientDSDMProgram = new PatientProgram();
+            newPatientDSDMProgram.setPatient(patient);
+            newPatientDSDMProgram.setDateEnrolled(session.getEncounter().getEncounterDatetime());
+            newPatientDSDMProgram.setProgram(getProgramByConceptFromObs(obsList));
+            newPatientDSDMProgram.setDateCompleted(null);
+            getProgramWorkflowService().savePatientProgram(newPatientDSDMProgram);
```

Comment:
@slubwama In some cases  you use getProgramWorkflowService() and in the other Context.getProgramWorkflowService() - use one consistently